Samples/SwiftUITestbed/Sources/RootWorkflow.swift

```diff
@@ -0,0 +1,111 @@
+ */
+
+import MarketWorkflowUI
+import Workflow
+import WorkflowUI
+
+struct RootWorkflow: Workflow {
+    let close: (() -> Void)?
+
+    typealias Output = Never
+
+    struct State {
+        var backStack: BackStack
+        var isPresentingModal: Bool
+
+        struct BackStack {
+            let root: Screen
+            var other: [Screen] = []
+        }
+
+        enum Screen {
+            case main(id: UUID = UUID())
+        }
+    }
+
+    func makeInitialState() -> State {
+        State(
+            backStack: .init(root: .main()),
+            isPresentingModal: false
+        )
+    }
+
+    enum Action: WorkflowAction {
+        typealias WorkflowType = RootWorkflow
+
+        case main(MainWorkflow.Output)
+        case popScreen
+        case dismissScreen
+
+        func apply(toState state: inout WorkflowType.State) -> WorkflowType.Output? {
+            switch self {
+            case .main(.pushScreen):
+                state.backStack.other.append(.main())
+            case .main(.presentScreen):
+                state.isPresentingModal = true
+            case .popScreen:
+                state.backStack.other.removeLast()
+            case .dismissScreen:
+                state.isPresentingModal = false
+            }
+            return nil
+        }
+    }
+
+    typealias Rendering = ModalContainer<BackStack, AnyScreen>
+    typealias BackStack = MarketBackStack<AnyMarketBackStackContentScreen>
+
+    func render(state: State, context: RenderContext<Self>) -> Rendering {
+        let sink = context.makeSink(of: Action.self)
+
+        func rendering(_ screen: State.Screen, isRoot: Bool) -> AnyMarketBackStackContentScreen {
+            switch screen {
+            case .main(let id):
+                return MainWorkflow(didClose: isRoot ? close : nil)
+                    .mapOutput(Action.main)
+                    .mapRendering(AnyMarketBackStackContentScreen.init)
+                    .rendered(in: context, key: id.uuidString)
+            }
+        }
+
+        func backStackContent(_ screen: State.Screen, isRoot: Bool) -> BackStack.Content {
+            rendering(screen, isRoot: isRoot)
+                .asContent(onPop: { sink.send(.popScreen) })
+        }
+
+        let backStack = MarketBackStack(
+            root: backStackContent(state.backStack.root, isRoot: true),
+            other: state.backStack.other.map { backStackContent($0, isRoot: false) }
+        )
+
+        return Rendering(
+            base: backStack,
+            modals: {
+                guard state.isPresentingModal else { return [] }
+                let screen = RootWorkflow(close: { sink.send(.dismissScreen) })
+                    .rendered(in: context)
+                    .asAnyScreen()
+                let modal = Modal(
+                    key: "",
+                    style: .full(),
+                    content: screen
+                )
+                return [modal]
+            }()
+        )
```

Comment:
IMO the preferred syntax here is:
```suggestion
        return MarketBackStack(
            root: backStackContent(state.backStack.root, isRoot: true),
            other: state.backStack.other.map { backStackContent($0, isRoot: false) }
        )
        .presentingModals {
            if state.isPresentingModal {
                RootWorkflow(close: { sink.send(.dismissScreen) })
                    .rendered(in: context)
                    .asAnyScreen()
                    .modal(
                        key: "modal",
                        style: .full()
                    )
            }
        }
```